### PR TITLE
First working state of the gem

### PR DESCRIPTION
**Pull request by galliani**
This PR brings the gem to a minimum working state, with now able to operate normally to fetch, send post or put request

Currently, only some of the modules have been covered with spec

### CODE REVIEW

File: lib/jurnal_api/client/journal_entries.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+module JurnalApi
+  class Client
+    # Defines methods related to SalesOrders
+    module JournalEntries
+
+      def journal_entries(params = {})
```
Comment (rslhdyt) [inline lib/jurnal_api/client/journal_entries.rb:6]: How about change naming for methods like this one.

```
def all
end

def find
end

def create
end

def update(id, params)
end

def delete(id)
```

File: lib/jurnal_api/client/journal_entries.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        response
+      end
+
+      def journal_entry_create(params = {})
+        response = post('journal_entries', params)
+        response        
+      end
+
+      def journal_entry_update(id)
+        response = delete("journal_entries/#{id}")
```
Comment (rslhdyt) [inline lib/jurnal_api/client/journal_entries.rb:22]: typo?

File: .env.template [hunk 3]
```diff
@@ -0,0 +1,2 @@
+JURNAL_COMPANY_API_KEY=
+JURNAL_API_URL=https://sandbox-api.jurnal.id
```
Comment (kikihakiem) [inline .env.template:2]: Why do we use environment variable for a gem configuration? I think it's better to use [configuration pattern](https://brandonhilkert.com/blog/ruby-gem-configuration-patterns/), so the user will have flexibility to configure. It's up to the user whether they want to use environment variable or some other technique.

File: spec/jurnal_api/client/sales_orders_spec.rb [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        @stubbed_id   = 1108 # from Jurnal API documentation
+        expected_url  = module_endpoint + '/' + @stubbed_id.to_s + '/convert_to_invoice.json'
+
+        @expected_stub = 
+          stub_request(:post, expected_url)
+            .with(body: dummy_params)
+            .to_return(status: 200, body: dummy_response)
+      end
+
+      subject { client.sales_order_convert_to_invoice(@stubbed_id, dummy_params) }
```
Comment (kikihakiem) [inline spec/jurnal_api/client/sales_orders_spec.rb:58]: use `subject!` if you want this subject to be eagerly evaluated

File: spec/jurnal_api/client/sales_orders_spec.rb [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+      it 'should hit the expected stub' do
+        subject
```
Comment (kikihakiem) [inline spec/jurnal_api/client/sales_orders_spec.rb:61]: so you don't need to do this

File: spec/jurnal_api/client/sales_orders_spec.rb [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        @expected_stub = 
+          stub_request(:post, module_endpoint + '.json')
+            .with(body: dummy_params)
+            .to_return(status: 201, body: dummy_response)
+      end
+
+      subject { client.sales_order_create(dummy_params) }
+
+      it 'should hit the expected stub' do
+        subject
```
Comment (kikihakiem) [inline spec/jurnal_api/client/sales_orders_spec.rb:28]: same for this

File: lib/jurnal_api/client/sales_orders.rb [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+      end
+
+      def sales_order_create(params = {})
+        response = post('sales_orders', params)
+        response        
+      end
+
+      def sales_order_convert_to_invoice(id, params = {})
+        url      = "sales_orders/#{id}/convert_to_invoice"
+        response = post(url, params)
```
Comment (kikihakiem) [inline lib/jurnal_api/client/sales_orders.rb:23]: unnecessary variable assignment?

File: lib/jurnal_api/client/journal_entries.rb (hunk 1 above)
Comment (kikihakiem) [reply lib/jurnal_api/client/journal_entries.rb:6]: agree. Prepending all the `JournalEntries` methods with `journal_entries` is redundant

File: jurnal-api.gemspec [hunk 8]
```diff
@@ -36,12 +36,18 @@ Gem::Specification.new do |spec|
...
-  spec.add_development_dependency "rspec", "~> 3.0"
-  spec.add_development_dependency "pry"
+  spec.add_development_dependency "bundler",          "~> 2.0"
+  spec.add_development_dependency "byebug",           "11.1.3"
+  spec.add_development_dependency "capybara",         "2.5.0"
+  spec.add_development_dependency "pry"  
+  spec.add_development_dependency "rake",             "~> 13.0"
+  spec.add_development_dependency "rspec",            "~> 3.0"
+  spec.add_development_dependency "shoulda-context",  "1.2.2"
+  spec.add_development_dependency "shoulda-matchers", "3.1.1"
```
Comment (kikihakiem) [inline jurnal-api.gemspec:46]: where is this gem being used?

**Pull request merged**